src/log.js

```diff
@@ -399,10 +399,11 @@ export class Log {
  * @return {string}
  */
 function toString(val) {
-  if (val instanceof Element) {
+  // Do check equivalent to `val instanceof Element` without cross-window bug
+  if (val.nodeType == 1) {
     return val.tagName.toLowerCase() + (val.id ? '#' + val.id : '');
   }
-  return val;
+  return /** @type {string} */ (val);
```

Comment:
The current code is equivalent to the code before, since Closure Compiler enforces `{!Element|string}` as the argument type, and the`if` statement above handles `Element`s, so below the `if` must be a `string`. So it would be technically unnecessary, but it is generally a good idea to be syntactically explicit. What do you think?